core/modules/replica/SqlRequest.cc

```diff
@@ -0,0 +1,334 @@
+    hdr.set_management_type(ProtocolManagementRequestType::REQUEST_STATUS);
+
+    buffer()->serialize(hdr);
+
+    ProtocolRequestStatus message;
+    message.set_id(id());
+    message.set_queued_type(ProtocolQueuedRequestType::SQL);
+
+    buffer()->serialize(message);
+
+    _send(lock);
+}
+
+
+void SqlRequest::_send(util::Lock const& lock) {
+
+    LOGS(_log, LOG_LVL_DEBUG, context() << __func__);
+
+    auto self = shared_from_base<SqlRequest>();
+
+    messenger()->send<ProtocolResponseSql>(
+        worker(),
+        id(),
+        buffer(),
+        [self] (string const& id,
+                bool success,
+                ProtocolResponseSql const& response) {
+
+            self->_analyze(success,
+                           response);
+        }
+    );
+}
+
+
+void SqlRequest::_analyze(bool success,
+                          ProtocolResponseSql const& message) {
+
+    LOGS(_log, LOG_LVL_DEBUG, context() << __func__ << "  success=" << (success ? "true" : "false"));
+
+    // This method is called on behalf of an asynchronous callback fired
+    // upon a completion of the request within method _send() - the only
+    // client of _analyze(). So, we should take care of proper locking and watch
+    // for possible state transition which might occur while the async I/O was
+    // still in a progress.
+
+    // IMPORTANT: the final state is required to be tested twice. The first time
+    // it's done in order to avoid deadlock on the "in-flight" callbacks reporting
+    // their completion while the request termination is in a progress. And the second
+    // test is made after acquiring the lock to recheck the state in case if it
+    // has transitioned while acquiring the lock.
+
+    if (state() == State::FINISHED) return;
+
+    util::Lock lock(_mtx, context() + __func__);
+
+    if (state() == State::FINISHED) return;
+
+    if (not success) {
+        finish(lock, CLIENT_ERROR);
+        return;
+    }
+
+    // Always use  the latest status reported by the remote server
+
+    setExtendedServerStatus(lock, replica::translate(message.status_ext()));
+
+    // Performance counters are updated from either of two sources,
+    // depending on the availability of the 'target' performance counters
+    // filled in by the 'STATUS' queries. If the later is not available
+    // then fallback to the one of the current request.
+
+    if (message.has_target_performance()) {
+        mutablePerformance().update(message.target_performance());
+    } else {
+        mutablePerformance().update(message.performance());
+    }
+
+    // Always extract extended data regardless of the completion status
+    // reported by the worker service.
+
+    _responseData.set(message);
+
+    // Extract target request type-specific parameters from the response
+    if (message.has_request()) {
+        _targetRequestParams = SqlRequestParams(message.request());
+    }
+    switch (message.status()) {
+
+        case ProtocolStatus::SUCCESS:
+
```

Comment:
the blank line here is inconsistent with other cases in the switch.

Reply:
👍